### PR TITLE
STOR-1359: Create StorageClass and manifest file in Azure File NFS job

### PR DESCRIPTION
**Pull request by bertinatto**

### CODE REVIEW

**Comment by bertinatto** [comment]
/pj-rehearse pull-ci-openshift-azure-file-csi-driver-operator-main-e2e-azure-file-nfs-csi

**Comment by bertinatto** [comment]
/pj-rehearse pull-ci-openshift-azure-file-csi-driver-operator-main-e2e-azure-file-nfs-csi

**Comment by bertinatto** [comment]
/pj-rehearse pull-ci-openshift-azure-file-csi-driver-operator-main-e2e-azure-file-nfs-csi

**Comment by bertinatto** [comment]
@openshift/storage PTAL

File: ci-operator/step-registry/storage/conf/csi-azure-file-nfs/storage-conf-csi-azure-file-nfs-commands.sh [hunk 1]
```diff
@@ -3,6 +3,33 @@ set -o errexit
...
+cat <<EOF >>$STORAGECLASS_LOCATION
+# Taken from https://github.com/kubernetes-sigs/azurefile-csi-driver/blob/master/deploy/example/storageclass-azurefile-nfs.yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: azurefile-csi-nfs
+provisioner: file.csi.azure.com
+parameters:
+  protocol: nfs
+  skuName: Premium_LRS
```
Comment (mpatlasov) [inline ci-operator/step-registry/storage/conf/csi-azure-file-nfs/storage-conf-csi-azure-file-nfs-commands.sh]: `assets/storageclass_nfs.yaml` from https://github.com/openshift/azure-file-csi-driver-operator doesn't have `skuName` and the test (`ci/prow/e2e-azure-file-nfs-csi`) is more-or-less happy w/o it. Do we consider the lack of `skuName` there as a bug and fix it here or there are other reasons for adding it?

Comment (bertinatto) [reply ci-operator/step-registry/storage/conf/csi-azure-file-nfs/storage-conf-csi-azure-file-nfs-commands.sh]: See a few lines above, this SC was taken from upstream, which is more or less a recommendation from MS.

The original SC you pointed above was also taken from upstream, but at the time it didn't have `skuName`, nor the `nsconnect` mount option.

**Comment by bertinatto** [comment]
/pj-rehearse pull-ci-openshift-azure-file-csi-driver-operator-main-e2e-azure-file-nfs-csi

**Comment by bertinatto** [comment]
I see both the test manifest and the SC created here:

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/40601/rehearse-40601-pull-ci-openshift-azure-file-csi-driver-operator-main-e2e-azure-file-nfs-csi/1674760076994285568/artifacts/e2e-azure-file-nfs-csi/storage-create-csi-azure-file-nfs/build-log.txt

Also, the rehease has passed:

/pj-rehearse ack

**Comment by bertinatto** [comment]
/pj-rehearse pull-ci-openshift-azure-file-csi-driver-master-e2e-azure-file-nfs-csi 	
/pj-rehearse pull-ci-openshift-azure-file-csi-driver-release-4.13-e2e-azure-file-nfs-csi
/pj-rehearse pull-ci-openshift-azure-file-csi-driver-release-4.12-e2e-azure-file-nfs-csi
/pj-rehearse pull-ci-openshift-azure-file-csi-driver-operator-main-e2e-azure-file-nfs-csi 	
/pj-rehearse pull-ci-openshift-azure-file-csi-driver-operator-release-4.13-e2e-azure-file-nfs-csi
/pj-rehearse pull-ci-openshift-azure-file-csi-driver-operator-release-4.12-e2e-azure-file-nfs-csi

**Pull request merged**